app/src/main/java/org/mozilla/fenix/library/history/HistoryController.kt

```diff
@@ -103,12 +103,12 @@ class DefaultHistoryController(
 
     override fun handleSearch() {
         val directions = if (settings.showUnifiedSearchFeature) {
-            HistoryFragmentDirections.actionGlobalSearchDialog(null)
+            NavGraphDirections.actionGlobalSearchDialog(null)
```

Comment:
I followed the assumption that global actions should be called from `NavGraphDirections`. Now that I have checked, we mostly use named directions, though some classes still use `NavGraphDirections`. What's the difference?

Reply:
From what I gather from looking at the code and reading https://developer.android.com/guide/navigation/navigation-getting-started#ensure_type-safety_by_using_safe_args, I believe the correct usage of `NavGraphDirections` is when we don't know the originating fragment such as deep intents. So, it's used as a global direction from unknown origins. Otherwise, I would suggest using the originating fragment direction. The action being global simply means we can call the action from any direction.

Reply:
@jonalmeida 
Yes, it could be called from both fragments. There is just want class `HistoryFragment`, but in the nav grath there are to fragments using it, with id `historyFragment` and `syncedHistoryFragment`.